packages/amplify_core/lib/src/types/models/auth_rule.dart

```diff
@@ -67,7 +67,7 @@ class AuthRule {
   }
 
   Map<String, dynamic> toMap() {
-    Map<String, dynamic> map = {
+    final map = <String, dynamic>{
```

Comment:
nit: `<String, dynamic>` probably is unnecessary the analyzer should be able to infer the type?

Reply:
discussed offline, but these changes were needed to satisfy the analysis linter as the files are now subject to analysis in core (but not datastore, yet).

Reply:
correction: @HuiSF is right about these maps that they appear to infer correctly if the `<String, dynamic>` removed (not the way it was in datastore plugin). will try to change them to infer as suggested

Reply:
fixed